pkg/image/importer/importer.go

```diff
@@ -537,6 +538,28 @@ func (isi *ImageStreamImporter) importRepositoryFromDocker(ctx gocontext.Context
 			manifest = m
 		}
 
+		if manifestList, isManifestList := manifest.(*manifestlist.DeserializedManifestList); isManifestList {
+			manifest = nil
+			for _, m := range manifestList.Manifests {
+				if m.Platform.OS == "linux" && m.Platform.Architecture == "amd64" {
```

Comment:
What about `runtime.GOOS` and `runtime.GOARCH` or maybe some other internal package defining what we support? At least, these could be constants taken from https://golang.org/pkg/runtime/internal/sys/#pkg-constants.

Reply:
https://github.com/docker/distribution/blob/06fa77aa11a3913096efcb9b5bd25db8ef55a939/registry/handlers/manifests.go#L25-L26

Reply:
I would prefer to read this from an annotation on the image stream tag and have it default to Linux / amd64.  We are very shortly going to need to do something better here for other reasons (multiarch), and if we need to work around for Windows we'll need the tag to be controlled.

The annotation should be a struct with arch and os called out as input.  If there is only one tag

Reply:
If there is only one tag, ignore the constraints for backwards compatibility.  Default the constraint to Linux/amd64

Reply:
Do we know for sure that all valid Linux amd64 images have both these values set correct from red hat or customer registries?

Reply:
@dmage bump ^^

if the answer is no, perhaps we update this PR to just strip out the windows manifests.

Reply:
You may still have other types.  As I mentioned, defaulting to linux/amd64 is fine, taking the first one if there is only one is fine, but we probably still need the escape hatch which is why the spec tag annotation API is probably going to be the safe fix.

Reply:
is the annotation api something we think is the long term solution to this?  I don't want to get stuck supporting that annotation 12 months from now when we've properly implemented manifestlist support.

(that annotation api would, at a minimum, i assume, have to support a the user providing a list of architectures to allow, not just one).